docs/ios/v2/how-to-guides/interact-with-room/room/polls.mdx

```diff
@@ -0,0 +1,155 @@
+
+// Create a quz
+let pollBuilder = HMSPollBuilder()
+    .withTitle("My Weather Quiz")
+    .withCategory(.quiz)
+    .addQuestion(with: questionBuilder)
+
+hmsSDK.interactivityCenter.quickStartPoll(with: pollBuilder) { success, error in
+		// handle error if any
+}
+```
+
+## Receiving Updates on a Poll
+
+Once you have start a poll, you can receive real-time updates on the poll's status and results using the `addPollUpdateListner` method in `HMSInteractivityCenter`. This method takes a closure as a parameter, which is called whenever an update occurs, such as when a poll starts, stops, or has its results updated.
+
+You can use the `HMSPoll` object returned by the closure to access information about the poll, such as its current state, duration, and questions. 
+
+```swift
+// Listen for poll updates 
+hmsSDK.interactivityCenter.addPollUpdateListner { [weak self] updatedPoll, update in
+  guard let self else { return }  
+		case .started:
+        self.showPollStartedToast()
+    case .resultsUpdated:
+        self.updateResultsScreen()
+    case .stopped:
+        self.loadResultsSummaryIfNeeded()
+    @unknown default:
+        break
+    }
+}
+```
+
+## Voting on a Poll
+
+To participate in a poll, you can use the `add(response:completion:)` method in `HMSInteractivityCenter`. This method takes a poll response builder object as a parameter, which you can use to add responses to the poll's questions. You can provide selected options or answers to short or long answer questions, depending on the question type.
+
+```swift
+let builder = HMSPollResponseBuilder(poll: poll)
+builder.addResponse(for: firstQuestion, options: selectedOption)
+hmsSDK.interactivityCetner.add(response: builder) { results, error in
+		// handle error if any
+}
+```
+
+Once you have submitted your responses, other peers in the room will get `resultsUpdated` update from the `addPollUpdateListner` callback. You can check updated vote counts for each option of poll questions:
+
+```swift
+hmsSDK.interactivityCenter.addPollUpdateListner { [weak self] updatedPoll, update in
+...
+		case .resultsUpdated:
+			print("Vote count for first option of first question is \(updatedPoll.questions?.first?.options?.first?.voteCount)")
+...
+}
+```
+
+You can use the `fetchResponses` method in `HMSInteractivityCenter` to fetch the detailed responses for the poll that will contain the information about the voter (in case of non anonymous poll) as well as responses to open ended questions that can’t be summarised. 
+
+## Stopping a Poll
+
+Use `stop(poll:completion:)` when you wan’t to prevent further responses recorded.
+
+## Viewing Poll Results Summary
+
+After poll has ended you can get a poll summary to get statistics like how many peers responded and how many total peers were in the room at the time of poll.  
+
+```swift
+hmsSDK.interactivityCenter.fetchPollResult(for: poll) { [weak self] poll, error in
+	let pollResult = poll?.result else { return }
```

Comment:
can i see the full list of params in the result? can we link to the corresponding API reference page?

Reply:
added a link to class reference